eox_nelp/course_experience/api/v1/serializers.py

```diff
@@ -13,6 +15,79 @@
 from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
 
 User = get_user_model()
+COURSE_OVERVIEW_EXTRA_FIELD_MAPPING = {"display_name": "display_name"}
+USER_EXTRA_FIELD_MAPPING = {
+    "first_name": "first_name",
+    "last_name": "last_name",
+    "profile__name": "profile_name",
+    "username": "username",
+}
+
+
+def get_course_extra_attributes(value=None):
+    """Function to retrieve CourseOverview extra fields
+
+    Args:
+        value (CourseOverview instance): CourseOverview that the relation analize. Defaults to None.
+
+    Returns:
+        dict: dict object too add course extra fields
+    """
+    return {
+        "attributes": map_attributes_from_instance_to_dict(value, COURSE_OVERVIEW_EXTRA_FIELD_MAPPING)
+    }
+
+
+def get_user_extra_attributes(value=None):
+    """Function to retrieve User extra fields
+
+    Args:
+        value (Userinstance): User that the relation analize. Defaults to None.
+
+    Returns:
+        dict: dict object too add user extra fields
+    """
+    return {
+        "attributes": map_attributes_from_instance_to_dict(value, USER_EXTRA_FIELD_MAPPING)
+    }
+
+
+def map_attributes_from_instance_to_dict(instance, attributes_mapping):
+    """Create a dictionary that represents some fields or attributes of a instance based on
+    a attributes_mapping dictionary. This dict would have key, values which the key represent the
+    attribute to look in the instance and the value the key name in the output dict.
+    Based in the `attributes_mapping` you should use a dict with the following config:
+    {
+        "field_name": "key_name"
+    }
+    This would check in the instace instance.field_name and the value send it to output dict
+    like {"key_name": instance.field_name}
+    Also its is possible to check nested fields if you declarate the field of instance separated by `__`
+    eg:
+    {
+        "field_level1__field_level2": "key_name"
```

Comment:
could you interchange  that,  I mean  {"key_name": "field_level1__field_level2"}, I don't have an specific use of that but if you interchange that it's possible to have different keys with same value 
```
{
  "key_name": "field_level1__field_level2",
  "key_2": "field_level1__field_level2"
},
```

Reply:
https://github.com/eduNEXT/eox-nelp/pull/64/commits/e371330e4d35fd33a0f1220ae6d6650dba3356c8